### PR TITLE
1-connect-to-db

### PR DESCRIPTION
**Pull request by alenaHalyha**

### CODE REVIEW

File: src/main/java/by/divergenny/traveler/domain/User.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    @Id
+    private String id;
+    private String firstName;
+    private String lastName;
+    private Date dateOfBirth;
+    private String country;
+    private String city;
+    private String about;
+
+    public User(String firstName, String lastName) {
```
Comment (ahalyha) [inline src/main/java/by/divergenny/traveler/domain/User.java]: Lets remake this constructor and add all other fields (except Id)

File: src/main/java/by/divergenny/traveler/mapper/UserMapper.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import org.mapstruct.Mapper;
+
+import java.util.List;
+
+@Mapper(componentModel = "spring")
+public interface UserMapper {
+    UserDto toDto(User entity);
+    User toEntity(UserDto dto);
+    List<UserDto> toDtos(List<User> entities);
+    List<User> toEntities(List<UserDto> dtos);
```
Comment (ahalyha) [inline src/main/java/by/divergenny/traveler/mapper/UserMapper.java]: not necessary to create this two methods, as we can convert list of User to UserDTO in stream for example

File: src/main/java/by/divergenny/traveler/dto/UserDto.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package by.divergenny.traveler.dto;
+
+import lombok.*;
+import lombok.experimental.SuperBuilder;
```
Comment (ahalyha) [inline src/main/java/by/divergenny/traveler/dto/UserDto.java]: unnecessary annotation

File: src/main/java/by/divergenny/traveler/persistence/UserRepository.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package by.divergenny.traveler.persistence;
+
+import by.divergenny.traveler.domain.User;
+import org.springframework.data.mongodb.repository.MongoRepository;
+
+import java.util.List;
+
+public interface UserRepository extends MongoRepository<User, String> {
```
Comment (ahalyha) [inline src/main/java/by/divergenny/traveler/persistence/UserRepository.java:10]: please add @Repository annotation here

File: src/main/java/by/divergenny/traveler/service/impl/UserServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        User userInSystem = userRepository.findById(userId).orElse(null);
+        if (null == userInSystem) {
+            return null;
+        }
+        checkAndSet(userInSystem, userWithNewData);
+        userRepository.save(userInSystem);
+        return userInSystem;
+    }
+
+    public void checkAndSet(User userInSystem, User userWithNewData) {
```
Comment (ahalyha) [inline src/main/java/by/divergenny/traveler/service/impl/UserServiceImpl.java]: lets make this method private

File: src/main/java/by/divergenny/traveler/persistence/UserRepository.java (hunk 4 above)
Comment (divergenny) [reply src/main/java/by/divergenny/traveler/persistence/UserRepository.java:10]: Done

File: src/main/java/by/divergenny/traveler/dto/UserDto.java (hunk 3 above)
Comment (divergenny) [reply src/main/java/by/divergenny/traveler/dto/UserDto.java]: Done

File: src/main/java/by/divergenny/traveler/mapper/UserMapper.java (hunk 2 above)
Comment (divergenny) [reply src/main/java/by/divergenny/traveler/mapper/UserMapper.java]: Done

File: src/main/java/by/divergenny/traveler/domain/User.java (hunk 1 above)
Comment (divergenny) [reply src/main/java/by/divergenny/traveler/domain/User.java]: Done

**Pull request merged**